nixos/modules/config/fonts/fonts.nix

```diff
@@ -69,13 +43,30 @@ in
         '';
       };
 
+      optimizeForVeryHighDPI = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          Optimize configuration for very high-density (>200 DPI) displays:
+            - disable subpixel anti-aliasing
+            - disable hinting
+            - automatically upscale the default X11 cursor
+        '';
+      };
     };
 
   };
 
   config = mkMerge [
-    { fonts.fonts = mkIf config.fonts.enableDefaultFonts defaultFonts; }
-    { fonts.fonts = mkIf config.services.xserver.enable defaultXFonts; }
+    { fonts.fonts = mkIf cfg.enableDefaultFonts defaultFonts; }
+    (mkIf cfg.optimizeForVeryHighDPI {
+      services.xserver.upscaleDefaultCursor = mkDefault true;
+      fonts.fontconfig = {
+        antialias = mkDefault false;
+        hinting.enable = mkDefault false;
+        subpixel.lcdfilter = mkDefault "none";
```

Comment:
FYI, my laptop's screen has 282 ppi (3840x2160, 15.6 inch, calculated using https://www.sven.de/dpi/), but I still need to set `antialias = true` and `subpixel.lcdfilter = "default"` to get a good font rendering. I cannot see the differences of different values of `hinting` and `rgba` though.

So I have the same question as @trofi https://github.com/NixOS/nixpkgs/pull/194594#issuecomment-1477665631:
> Are you sure disabling aliasing is a reasonable thing to do?

Reply:
It's the recommended default (both in the NixOS manual and various online sources, see original PR discussion) and it seems to work best for *most* users.

At the end of the day, font rendering must conform to individual users' needs and subjective experiences (no two persons see and interpret written text quite the same way) so making it an easily-overridden default was pretty deliberate.
Of course if there's good evidence antialiasing is better for most users, we should have it on.

A good value for `lcdfilter` is harder to "guess" : something other than `none` should match the LCD panel's orientation, and even panels integrated into laptops etc. may be mounted sideways; it seems not to be uncommon, since designers can then use panels originally meant for use in tablets.  :grimacing: